server/src/main/java/org/elasticsearch/ingest/CompoundProcessor.java

```diff
@@ -133,30 +133,45 @@ void innerExecute(int currentProcessor, IngestDocument ingestDocument, BiConsume
         final IngestMetric metric = processorWithMetric.v2();
         final long startTimeInNanos = relativeTimeProvider.getAsLong();
         metric.preIngest();
-        processor.execute(ingestDocument, (result, e) -> {
-            long ingestTimeInNanos = relativeTimeProvider.getAsLong() - startTimeInNanos;
-            metric.postIngest(ingestTimeInNanos);
+        try {
+            processor.execute(ingestDocument, (result, e) -> {
+                long ingestTimeInNanos = relativeTimeProvider.getAsLong() - startTimeInNanos;
+                metric.postIngest(ingestTimeInNanos);
```

Comment:
Should we be calling this in case of failure as well?

Reply:
specifically `metric.postIngest(ingestTimeInNanos);`

\me shakes fist at IDEA review tools picking four lines for the comment

Reply:
Good point. I think `metric.postIngest(ingestTimeInNanos);` should be called as well when catch exception at line 151. 

(in case an error comes in via this handler (meaning `e` is not null) then this is already invoked)